--- memcached-1.4.5/items.c.warnings
+++ memcached-1.4.5/items.c
@@ -449,10 +449,8 @@ void do_item_stats_sizes(ADD_STAT add_st
         /* write the buffer */
         for (i = 0; i < num_buckets; i++) {
             if (histogram[i] != 0) {
-                char key[8];
-                int klen = 0;
-                klen = snprintf(key, sizeof(key), "%d", i * 32);
-                assert(klen < sizeof(key));
+                char key[12];
+                snprintf(key, sizeof(key), "%d", i * 32);
                 APPEND_STAT(key, "%u", histogram[i]);
             }
         }
--- memcached-1.4.5/memcached.c.warnings
+++ memcached-1.4.5/memcached.c
@@ -4662,9 +4662,9 @@ int main (int argc, char **argv) {
 
         /* create the UDP listening socket and bind it */
         errno = 0;
-        if (settings.udpport && server_socket(settings.udpport, udp_transport,
-                                              portnumber_file)) {
-            vperror("failed to listen on UDP port %d", settings.udpport);
+        if (udp_port && server_socket(udp_port, udp_transport,
+                                      portnumber_file)) {
+            vperror("failed to listen on UDP port %d", udp_port);
             exit(EX_OSERR);
         }
 
